### DISCUSSION TITLE
Image Magick 6.9.11.60 Does not Open ARW Sony RAW Image File

### DISCUSSION THREAD

**Discussion by CaptainMorgan12** [ImageMagick/ImageMagick#7113] [General] [answered]
Mime Type: x-sony-arw

Using the GUI I am getting an error:
`Unknown field with tag 50341 (0xc4a5) encountered. 'TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/985:`

Since I am trying to use Image Magick for my RAW thumbnailer files in nautilus these are not generated. It works perfect for my RAW DNG, RAF and other  RAW image files, just not ARW.

The script I am using has the following packages to use imagemagick mostly to help with converting RAF to jpg / png thumbnails. Is there a package needed for ARW?

[Source](https://discuss.pixls.us/t/raf-and-xcf-thumbnails-in-ubuntu-20-04/22898/6)

>  - gridsite-clients for urlencode
>  - exiv2 for exiv2
>  - imagemagick for convert`

**Comment by snibgo** [comment]
> warning/tiff.c/TIFFWarnings/985

That is a warning, not an error.

What is the text output from `convert -version`?

ARW has multiple formats. Perhaps the raw reader can't read whatever version you have. If you zip a typical ARW and post that here, we can take a look.

**Reply by CaptainMorgan12** [reply]
Sorry the files are 85MB so can't attach. I compiled Image Magick from source now running ImageMagick 7.1.1-29, ~~ARW files do not open properly,~~ still generates the same error as well in the GUI:

`magick open /home/user/Pictures/"Mountains Outside"/_DSC1112.ARW
magick: no decode delegate for this image format `' @ error/constitute.c/ReadImage/746.
`

`<pre>convert -version
Version: ImageMagick 7.1.1-29 (Beta) Q16-HDRI x86_64 3132d7884:20240214 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(4.5) 
Delegates (built-in): bzlib djvu fontconfig freetype heic jbig jng jp2 jpeg jxl lcms lqr ltdl lzma pangocairo png raqm raw tiff webp wmf x xml zip zlib zstd
Compiler: gcc (13.2)
</pre>`

Using 

`sudo convert ~/Pictures/Outside/_DSC0149.ARW -resize 50% ~/Pictures/Outside/small.jpg
`
Interestingly works and it generates the jpg file

Ok I now know what is happening it is converting the ARW files and thumbnails are generated. I left the folder open over night and suddenly all the ARW thumbnails were there. **The issue now is that it is slow**, so maybe a different issue.

Some image info:

`identify ~/Pictures/Outside/_DSC0149.ARW
/home/amiga/Pictures/Outside/_DSC0149.ARW ARW 7968x5320 8000x5320+0+0 16-bit sRGB 81.6934MiB 0.000u 0:00.001
`

`$ identify -verbose ~/Pictures/outside/_DSC0149.ARW
Image:
  Filename: /home/amiga/Pictures/outside/_DSC0149.ARW
  Permissions: rw-rw-r--
  Format: ARW (Sony Alpha Raw Image Format)
  Class: DirectClass
  Geometry: 7968x5320+0+0
  Units: Undefined
  Colorspace: sRGB
  Type: TrueColor
  Base type: Undefined
  Endianness: Undefined
  Depth: 16-bit
  Channels: 3.0
  Channel depth:
    Red: 16-bit
    Green: 16-bit
    Blue: 16-bit
  Channel statistics:
    Pixels: 42389760
    Red:
      min: 0  (0)
      max: 65535 (1)
      mean: 17415 (0.265735)
      median: 16180 (0.246891)
      standard deviation: 11898.9 (0.181566)
      kurtosis: 3.99193
      skewness: 1.60825
      entropy: 0.893913
    Green:
      min: 0  (0)
      max: 65535 (1)
      mean: 23854.6 (0.363997)
      median: 25451 (0.388357)
      standard deviation: 12805 (0.195392)
      kurtosis: 0.286279
      skewness: 0.386507
      entropy: 0.935444
    Blue:
      min: 0  (0)
      max: 65535 (1)
      mean: 32791.8 (0.50037)
      median: 30943 (0.47216)
      standard deviation: 17087 (0.260731)
      kurtosis: -1.30991
      skewness: -0.188158
      entropy: 0.962636
  Image statistics:
    Overall:
      min: 0  (0)
      max: 65535 (1)
      mean: 24687.1 (0.376701)
      median: 24191.3 (0.369136)
      standard deviation: 13930.3 (0.212563)
      kurtosis: 0.989433
      skewness: 0.602201
      entropy: 0.930664
  Rendering intent: Perceptual
  Gamma: 0.454545
  Chromaticity:
    red primary: (0.64,0.33,0.03)
    green primary: (0.3,0.6,0.1)
    blue primary: (0.15,0.06,0.79)
    white point: (0.3127,0.329,0.3583)
  Matte color: grey74
  Background color: white
  Border color: srgb(223,223,223)
  Transparent color: black
  Interlace: None
  Intensity: Undefined
  Compose: Over
  Page geometry: 8000x5320+0+0
  Dispose: Undefined
  Iterations: 0
  Compression: Undefined
  Orientation: Undefined
  Properties:
    date:create: 20##-02-11T04:42:17+00:00
    date:modify: 20##-08-26T21:52:06+00:00
    date:timestamp: 20##-02-16T18:25:40+00:00
    dng:camera.model.name: DSC-RX1RM2
    dng:create.date: 2017-08-26T21:52:07+00:00
    dng:exposure.time: 1/1e+03
    dng:f.number: 2
    dng:focal.length: 4e+01 mm
    dng:focal.length.in.35mm.format: 35 mm
    dng:gps.altitude: 0 m
    dng:gps.latitude: 0 deg 0' 0" N
    dng:gps.longitude: 0 deg 0' 0" W
    dng:iso.setting: 5e+01
    dng:lens: 4e+01-0mm f/2-0
    dng:lens.f.stops: 0.00
    dng:lens.type: 
    dng:make: Sony
    dng:max.aperture.at.max.focal: 0
    dng:max.aperture.at.min.focal: 2
    dng:max.aperture.value: 2
    dng:max.focal.length: 0 mm
    dng:min.focal.length: 35.0 mm
    dng:software: DSC-RX1RM2 v1.00
    dng:wb.rb.levels: 2480 1688 1024 1024
    signature: 8b797146ab8225308734d76398fa7872ef54abf6017e95efae2c130c13040a6e
  Artifacts:
    verbose: true
  Tainted: False
  Filesize: 81.6934MiB
  Number pixels: 42.3898M
  Pixel cache type: Memory
  Pixels per second: 10.7693MP
  User time: 9.760u
  Elapsed time: 0:04.936
  Version: ImageMagick 7.1.1-29 (Beta) Q16-HDRI x86_64 3132d7884:20240214 https://imagemagick.org
`

**Reply by fmw42** [reply, marked as answer]
> magick open /home/user/Pictures/"Mountains Outside"/_DSC1112.ARW

There is no Imagemagick command called "open".  If you want to convert it, then 

`convert image.suffix result.new_suffix`

If you want to view it on Unix-like system, then

`display image.suffix`

**Reply by CaptainMorgan12** [reply]
`display image.ARW` works

**Reply by fmw42** [reply]
See https://imagemagick.org/script/sitemap.php#command-line for all the command line tool names

**Comment by snibgo** [comment]
> magick open /home/user/Pictures/"Mountains Outside"/_DSC1112.ARW

As fmw42 says, that command is bad. It tells `magick` to read a file called `open`.

**Reply by CaptainMorgan12** [reply]
Understood thanks for the link and tips! Feel free to close. The gui Image Magick 6.9.11.60 still generates the error and I did not compile that from source, or know if there is a newer version. But magick itself seems to work fine. I noticed that for some folders where it did not show the .ARW files thumbnails and didn't start generating them even when waiting several hours issuing the command

$display _DSC5962.ARW 

in that folder for any ARW image kicked off the thumbnailing process. However just entering the folder in Nautilus did not, not exactly sure why it wouldn't automatically do it.
